instrumentation/net_http/lib/opentelemetry/instrumentation/net/http/patches/instrumentation.rb

```diff
@@ -11,20 +11,35 @@ module HTTP
         module Patches
           # Module to prepend to Net::HTTP for instrumentation
           module Instrumentation
+            include Common::HTTP::RequestAttributes
+
             HTTP_METHODS_TO_SPAN_NAMES = Hash.new { |h, k| h[k] = "HTTP #{k}" }
             USE_SSL_TO_SCHEME = { false => 'http', true => 'https' }.freeze
 
-            def request(req, body = nil, &block)
+            def request(req, body = nil, &block) # rubocop:disable Metrics/AbcSize, Metrics/MethodLength
               # Do not trace recursive call for starting the connection
               return super(req, body, &block) unless started?
 
-              attributes = {
-                OpenTelemetry::SemanticConventions::Trace::HTTP_METHOD => req.method,
-                OpenTelemetry::SemanticConventions::Trace::HTTP_SCHEME => USE_SSL_TO_SCHEME[use_ssl?],
-                OpenTelemetry::SemanticConventions::Trace::HTTP_TARGET => req.path,
-                OpenTelemetry::SemanticConventions::Trace::NET_PEER_NAME => @address,
-                OpenTelemetry::SemanticConventions::Trace::NET_PEER_PORT => @port
-              }.merge!(OpenTelemetry::Common::HTTP::ClientContext.attributes)
+              method = req.method
+
+              options = if req.uri
```

Comment:
Might be worth explaining why a req may/may not have `.uri` in a comment here

Reply:
https://github.com/open-telemetry/opentelemetry-ruby/pull/946#discussion_r716972224 will add a comment